b2/transferer.py

```diff
@@ -97,6 +118,70 @@ def _download_file_simple(self, file, response):
             bytes_read += len(data)
         return bytes_read, digest.hexdigest()
 
+    def _download_file_parallel(
+        self,
+        file,
+        response,
+        metadata,
+        max_streams=DEFAULT_MAX_STREAMS,
+        min_part_size=DEFAULT_MIN_PART_SIZE,
+    ):
+        """
+        Downloads a file from given url using parallel download sessions and stores it in the given download_destination.
+
+        Returns a dict containing all of the file info from the headers in the reply.
+
+        :param file: an opened file-like object to write to
+        :param response: The response of the first request made to the cloud service with download intent
+        :param max_streams: maximum number of simultaneous streams
+        :param min_part_size: minimum amount of data a single stream will retrieve, in bytes
+        :return:
+        """
+        assert max_streams >= 1
+
+        info_dict = FileMetadata.from_response(response).as_info_dict()
+
+        content_length = info_dict['contentLength']
+        number_of_streams = min(max_streams, content_length // min_part_size)
+        part_size = content_length // number_of_streams
```

Comment:
What happens if the content length is not a multiple of the number of streams?  The parts can't all be the same size.

Reply:
I think the `_get_parts` method is asking for ranges that are one byte bigger that `part_size`, which would solve the problem.  It's probably worth some comments to make it clearer.

Reply:
I'll fix it